dspace_rest_client/client.py

```diff
@@ -1149,18 +1149,47 @@ def get_item(self, uuid, embeds=None):
         """
         Get an item, given its UUID
         @param uuid:    the UUID of the item
-        @param embeds:      Optional list of resources to embed in response JSON
-        @return:        the raw API response
+        @param embeds:  Optional list of resources to embed in response JSON
+        @return:        the constructed Item object or None if an error occurs
         """
-        # TODO - return constructed Item object instead, handling errors here?
         url = f"{self.API_ENDPOINT}/core/items"
         try:
-            id = UUID(uuid).version
+            # Validate the UUID format
+            id_version = UUID(uuid).version
             url = f"{url}/{uuid}"
-            return self.api_get(url, parse_params(embeds=embeds), None)
+
+            # Make API GET request
+            response = self.api_get(url, parse_params(embeds=embeds), None)
+
+            # Handle successful response
+            if response.status_code == 200:
+                # Parse the response JSON into an Item object
+                return self._construct_item(response.json())
+            else:
+                logging.error(
+                    "Failed to retrieve item. Status code: %s", response.status_code
+                )
+                logging.error("Response: %s", response.text)
+                return None
         except ValueError:
             logging.error("Invalid item UUID: %s", uuid)
             return None
+        except Exception as e:
+            logging.error("An unexpected error occurred: %s", str(e))
+            return None
+
+    def _construct_item(self, item_data):
+        """
+        Construct an Item object from API response data
+        @param item_data: The raw JSON data from the API
+        @return: An Item object
+        """
+        try:
+            # Create an Item instance, using the API response data
+            return Item(api_resource=item_data)
```

Comment:
Given all this does is one line, I'm wondering what the advantage of this function is... the auto exception handling?

My thinking when using the constructors like this is that it lets us be a bit more abstract, and do things like

```
dso_type = type(ds)
...
return dso_type(api_resource=parse_json(r))
```

Does that make sense? Do you think we need to just be more strict about handling each different DSO type? Or, if the error handling is the point, could we instead change this to _construct_dso(self, dso_data, dso_type) ?